linera-service/src/linera/main.rs

```diff
@@ -1481,55 +1477,80 @@ async fn run(options: ClientOptions) -> Result<(), anyhow::Error> {
-                            initial_amount: Amount::from_tokens(*initial_amount),
-                            num_initial_validators: *validators,
-                            num_shards: *shards,
-                            binaries: binaries.clone().into(),
-                            policy: ResourceControlPolicy::default(),
-                        };
-                        let (mut net, client1) = config.instantiate().await?;
-                        let result = Ok(net_up(extra_wallets, &mut net, client1).await?);
-                        listen_for_signals(&mut shutdown_receiver, &mut net).await?;
-                        result
-                    }
-                    #[cfg(not(feature = "kubernetes"))]
-                    bail!("Cannot use the kubernetes flag with the kubernetes feature off")
-                } else {
-                    let tmp_dir = tempdir()?;
-                    let path = tmp_dir.path();
-                    let path_buf = path.to_path_buf();
-                    let storage_config = StorageConfig::RocksDb { path: path_buf };
-                    let storage_config_builder =
-                        StorageConfigBuilder::ExistingConfig { storage_config };
-                    let path_provider = PathProvider::new(path);
-                    let config = LocalNetConfig {
-                        network: Network::Grpc,
-                        database: Database::RocksDb,
-                        testing_prng_seed: *testing_prng_seed,
-                        table_name: table_name.to_string(),
-                        num_other_initial_chains: *other_initial_chains,
-                        initial_amount: Amount::from_tokens(*initial_amount),
-                        num_initial_validators: *validators,
-                        num_shards: *shards,
-                        policy: ResourceControlPolicy::default(),
-                        storage_config_builder,
-                        path_provider,
-                    };
-                    let (mut net, client1) = config.instantiate().await?;
-                    let result = Ok(net_up(extra_wallets, &mut net, client1).await?);
-                    listen_for_signals(&mut shutdown_receiver, &mut net).await?;
-                    result
+            #[cfg(feature = "rocksdb")]
+            NetCommand::Up {
+                extra_wallets,
+                other_initial_chains,
+                initial_amount,
+                validators,
+                shards,
+                testing_prng_seed,
+                table_name,
+                ..
+            } => {
+                use linera_service::{
+                    cli_wrappers::{
+                        local_net::{Database, LocalNetConfig, PathProvider, StorageConfigBuilder},
+                        LineraNetConfig, Network,
+                    },
+                    storage::StorageConfig,
+                };
+
```

Comment:
Optional: I'd prefer to have this import (and the other one above) feature gated at the top of the file. That keeps them all in one place and makes it easier to see that this module depends on these imports and may be affected the `rocksdb` and `kubernetes` feature flags :thinking: